### ISSUE TITLE
Failed for format warning

### ISSUE THREAD

**Issue by kipcole9** [jeremyjh/dialyxir#172]
Thanks for the speedy update.  I ran dialyzer again and as prompted here is the current format failure:
```elixir
lib/money.ex:1194:call
The call:
Money.to_currency(
  _any(),
  _any(),
  {:error, {%{:__exception__ => true, :__struct__ => atom(), atom() => _}, binary()}}
  | {:ok, map()}
)

breaks the contract
(Money.t(), currency_code(), map()) ::
  {:ok, Money.t()} | {:error, {Exception.t(), String.t()}}
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(_money@1 :: %Money{:currency => _, _ => _}, _to_currency@1 :: atom() | binary(), {:ok, map()}) :: :ok\ndef a() do\n  :ok\nend\n"
```

**Comment by asummers** [comment]
Is there more to the second one with the original warning? That _should_ be working but clearly isn't. First one looks great now =)

**Comment by kipcole9** [comment]
@asummers I'm not 100% certain I'm posting the right error text - am assuming the messages between the dashed-lines is the correct way to do so?

BTW, this new formatting is the first time I've been able to make headway on Dialyzer - seriously great work and hence very much appreciated.

On current master I am getting the following on `ex_money`:
```elixir
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(_flows@1 :: nonempty_maybe_improper_list(), _estimate_m@1 :: float(), _estimate_n@1 :: float()) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/money/financial.ex:236: The call 'Elixir.Money.Financial':do_internal_rate_of_return(_flows@1::nonempty_maybe_improper_list(),_estimate_m@1::float(),_estimate_n@1::float()) will never return since it differs in the 2nd and 3rd argument from the success typing arguments: ([{integer(),#{'__struct__':='Elixir.Money', 'amount':=map(), 'currency':=atom()}},...],integer(),integer())
```
The second issue from yesterday was a syntax error on `ex_cldr` which I am no longer getting.  I am getting several errors of which the following are a few.  I am hoping (for your sake 😄 ) that they are the same issue.
```elixir
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(<:few | :many | :one | :other | :two | {:error, {Cldr.InvalidLanguageTag, <<_ :: 64, _ :: size(8)>>} | {Cldr.UnknownPluralRules, <<_ :: 64, _ :: size(8)>>}}, non_neg_integer()>) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/gettext/plural.ex:85: The pattern <'zero', __n@1> can never match the type <'few' | 'many' | 'one' | 'other' | 'two' | {'error',{'Elixir.Cldr.InvalidLanguageTag',<<_:64,_:_*8>>} | {'Elixir.Cldr.UnknownPluralRules',<<_:64,_:_*8>>}},non_neg_integer()>
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(%Decimal{:coef => :inf | :qNaN | :sNaN | non_neg_integer(), :exp => integer(), :sign => -1 | 1}, _n@1 :: any()) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:263: The call 'Elixir.Cldr.Math':round_significant(#{'__struct__':='Elixir.Decimal', 'coef':='inf' | 'qNaN' | 'sNaN' | non_neg_integer(), 'exp':=integer(), 'sign':=-1 | 1},_n@1::any()) will never return since it differs in the 1st argument from the success typing arguments: (number(),any())
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(_number@1 :: %Decimal{:sign => _, _ => _}) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:270: The call 'Elixir.Cldr.Math':log10(_number@1::#{'__struct__':='Elixir.Decimal', 'sign':=_, _=>_}) will never return since it differs in the 1st argument from the success typing arguments: (number())
________________________________________________________________________________
```

**Comment by asummers** [comment]
> am assuming the messages between the dashed-lines is the correct way to do so?

Yep! Thank you =) These are helpful. 

> I am hoping (for your sake 😄 ) that they are the same issue.

Two look to be the same, so good news in that department =)

**Comment by asummers** [comment]
All easy enough fixes. Should be in master soon.

**Comment by kipcole9** [comment]
Looking good on the latest changes - a lot less formatting errors.  But still a few.  I apologies that its a long list - but it is the full list of errors on `ex_cldr` so hopefully that accounts for something!

```elixir
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(<:few | :many | :one | :other | :two | {:error, {Cldr.InvalidLanguageTag, <<_ :: 64, _ :: size(8)>>} | {Cldr.UnknownPluralRules, <<_ :: 64, _ :: size(8)>>}}, non_neg_integer()>) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/gettext/plural.ex:85: The pattern <'zero', __n@1> can never match the type <'few' | 'many' | 'one' | 'other' | 'two' | {'error',{'Elixir.Cldr.InvalidLanguageTag',<<_:64,_:_*8>>} | {'Elixir.Cldr.UnknownPluralRules',<<_:64,_:_*8>>}},non_neg_integer()>
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(%Decimal{:coef => :inf | :qNaN | :sNaN | non_neg_integer(), :exp => integer(), :sign => -1 | 1}, _n@1 :: any()) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:263: The call 'Elixir.Cldr.Math':round_significant(#{'__struct__':='Elixir.Decimal', 'coef':='inf' | 'qNaN' | 'sNaN' | non_neg_integer(), 'exp':=integer(), 'sign':=-1 | 1},_n@1::any()) will never return since it differs in the 1st argument from the success typing arguments: (number(),any())
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(_number@1 :: %Decimal{:sign => _, _ => _}) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:270: The call 'Elixir.Cldr.Math':log10(_number@1::#{'__struct__':='Elixir.Decimal', 'sign':=_, _=>_}) will never return since it differs in the 1st argument from the success typing arguments: (number())
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(_sqrt_mantissa@1 :: float(), %Decimal{:coef => 1, :exp => 0, :sign => 1}) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:319: The call 'Elixir.Decimal':sub(_sqrt_mantissa@1::float(),#{'__struct__':='Elixir.Decimal', 'coef':=1, 'exp':=0, 'sign':=1}) will never return since the success typing is (binary() | integer() | #{'__struct__':='Elixir.Decimal', _=>_},binary() | integer() | #{'__struct__':='Elixir.Decimal', _=>_}) -> #{'__struct__':='Elixir.Decimal', 'coef':='inf' | 'qNaN' | 'sNaN' | non_neg_integer(), 'exp':=integer(), 'sign':=-1 | 1} and the contract is (decimal(),decimal()) -> t()
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(_number@1 :: %Decimal{_ => _}) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:369: The call 'Elixir.Cldr.Math':log(_number@1::#{'__struct__':='Elixir.Decimal', _=>_}) will never return since it differs in the 1st argument from the success typing arguments: (number())
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(<number(), pos_integer(), float()>) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/helpers/math.ex:687: The pattern <_number@1 = #{'__struct__':='Elixir.Decimal'}, _nth@1 = #{'__struct__':='Elixir.Decimal'}, _root@1 = #{'__struct__':='Elixir.Decimal'}> can never match the type <number(),pos_integer(),float()>
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a([{atom(), _}, ...], _param@2 :: binary()) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
lib/cldr/plug/plug_set_locale.ex:262: The call 'Elixir.Cldr.Plug.SetLocale':validate_from([{atom(),_},...],_param@2::binary()) will never return since it differs in the 2nd argument from the success typing arguments: ('nil' | [{atom(),_}] | map(),atom() | maybe_improper_list())
________________________________________________________________________________
Please file a bug in https://github.com/jeremyjh/dialyxir/issues with this message.

Failed to format warning:
"@spec a(<:equals, _, _>) :: :ok\ndef a() do\n  :ok\nend\n"

Legacy warning:
src/plural_rules_parser.yrl:129: The pattern <{'mod', [_C], [{_L}]}, A, B> can never match the type <'equals',_,_>
________________________________________________________________________________

```

**Comment by asummers** [comment]
Wait you're getting those with the changes in the PR? They haven't been merged yet. Strange if so, I have an explicit test case for the top error.

**Comment by kipcole9** [comment]
Maybe I jumped the gun - if so my apologies.  I updated deps and there were some commit(s) that were pulled so I assumed it was related to your PR above.  My bad.  I'll wait a while (was just trying to use time effectively during a flight delay)

**Comment by asummers** [comment]
I do see some errors in that project, now that I have it running on latest though.

**Comment by asummers** [comment]
@kipcole9 after #175 gets merged, the rest will be duplicates of #165. 

If you're looking for an opportunity to contribute, there's no explanation yet for `:call_without_opaque` and your project has that, and its formatting is a little lacking. PRs are definitely welcome once you figure out what's happening there =)

**Comment by kipcole9** [comment]
Looks great - no format errors on `ex_cldr` or `ex_money`.  But ..... a ton of work to do to fix the warnings. Which, thanks to your work, I may now be able to work through.  Happy to contribute when I get that part done since presumably I may even know what I'm doing by then.  Thanks again!
